pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12137079?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12137079?buildTab=artifacts#/) on release-23.2 @ [437fa1b407bb5078a2a98c0fb5b8e5436a3be926](https://github.com/cockroachdb/cockroach/commits/437fa1b407bb5078a2a98c0fb5b8e5436a3be926):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/cast (1m10s)

```
Stack: 

```
goroutine 7145559 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:98: Executing "SELECT d::DECIMAL(10,1) FROM t45837"
    comparator.go:98: Executing "CREATE TABLE t2 AS SELECT 18446744073709551616::FLOAT8 AS f"
    comparator.go:98: Executing "SELECT f::INT8 FROM t2"
    comparator.go:98: Executing "SELECT 23414123414::INT2"
    comparator.go:98: Executing "CREATE TABLE t0 (c0 DECIMAL UNIQUE); INSERT INTO t0(c0) VALUES (0)"
    comparator.go:98: Executing "CREATE TABLE t1 (c0 DECIMAL); INSERT INTO t1(c0) VALUES (0)"
    comparator.go:98: Executing "SELECT t0.c0 FROM t0 WHERE t0.c0 BETWEEN t0.c0 AND '-1'::INTERVAL::DECIMAL"
    comparator.go:98: Executing "SELECT t1.c0 FROM t1 WHERE t1.c0 BETWEEN t1.c0 AND '-1'::INTERVAL::DECIMAL"
    comparator.go:98: Executing "CREATE TABLE t64429 (_int8 INT8, _int4 INT4)"
    comparator.go:98: Executing "INSERT INTO t64429 VALUES (3000000000, 300000)"
    comparator.go:98: Executing "SELECT _int8::INT2 FROM t64429"
    comparator.go:98: Executing "SELECT _int8::INT4 FROM t64429"
    comparator.go:98: Executing "SELECT _int4::INT2 FROM t64429"
    comparator.go:98: Executing "DELETE FROM t64429 WHERE true; INSERT INTO t64429 VALUES (-3000000000, -300000)"
    comparator.go:98: Executing "SELECT _int8::INT2 FROM t64429"
    comparator.go:98: Executing "SELECT _int8::INT4 FROM t64429"
    comparator.go:98: Executing "SELECT _int4::INT2 FROM t64429"
    comparator.go:98: Executing "CREATE TABLE t66067_a (a INT8, c CHAR(26), INDEX (c, a))"
    comparator.go:98: Executing "CREATE TABLE t66067_b (a INT8, v VARCHAR(40))"
    comparator.go:98: Executing "INSERT INTO t66067_a VALUES (1, 'foo'); INSERT INTO t66067_b VALUES (1, 'bar')"
    comparator.go:98: Executing "SELECT * FROM t66067_b AS b INNER LOOKUP JOIN t66067_a AS a ON b.a = a.a WHERE (b.v = 'bar') AND (a.c = 'foo')"
    comparator.go:98: Executing "SELECT i, i::\"char\"::BYTES, length(i::\"char\") FROM (VALUES (32), (97), (127), (0), (-1), (-127), (-128)) AS v (i)"
    comparator.go:98: Executing "SELECT 128::\"char\""
    comparator.go:98: Executing "SELECT (-129)::\"char\""
    comparator.go:98: Executing "SELECT ' 1 '::INT8, ' 1.2 '::FLOAT8, ' 2.3 '::DECIMAL, ' true '::BOOL"
    comparator.go:98: Executing "SELECT i, i::OID, i::OID::STRING, i::OID::REGPROC, i::OID::REGPROCEDURE, i::OID::REGNAMESPACE, i::OID::REGCLASS, i::OID::REGTYPE, i::OID::REGROLE FROM (VALUES (0), (1)) AS v (i)"
    comparator.go:98: Executing "SELECT i, i::REGPROC::OID, i::REGPROCEDURE::OID, i::REGNAMESPACE::OID, i::REGTYPE::OID, i::REGCLASS::OID, i::REGROLE::OID, i::REGPROC, i::REGPROCEDURE, i::REGNAMESPACE, i::REGTYPE, i::REGCLASS, i::REGROLE FROM (VALUES ('-')) AS v (i)"
    comparator.go:98: Executing "SELECT i, i::OID FROM (VALUES ('-')) AS v (i)"
    comparator.go:98: Executing "SELECT '-'::OID"
    comparator.go:98: Executing "SELECT '-'::REGCLASS, '-'::REGCLASS::OID, '-'::REGPROC, '-'::REGPROC::OID, '-'::REGPROCEDURE, '-'::REGPROCEDURE::OID, '-'::REGNAMESPACE, '-'::REGNAMESPACE::OID, '-'::REGTYPE, '-'::REGTYPE::OID, '-'::REGROLE, '-'::REGROLE::OID"
    comparator.go:98: Executing "PREPARE s73450_a AS SELECT $1::INT2"
    comparator.go:98: Executing "EXECUTE s73450_a (999999)"
    comparator.go:98: Executing "PREPARE s73450_b AS SELECT $1::CHAR"
    comparator.go:98: Executing "EXECUTE s73450_b ('foo')"
    comparator.go:98: Executing "CREATE TABLE t73450 (c CHAR)"
    comparator.go:98: Executing "INSERT INTO t73450 VALUES ('f')"
    comparator.go:98: Executing "SELECT * FROM t73450 WHERE c = 'foo'::CHAR"
    comparator.go:98: Executing "PREPARE s73450_c AS SELECT * FROM t73450 WHERE c = $1::CHAR"
    comparator.go:98: Executing "EXECUTE s73450_c ('foo')"
    comparator.go:98: Executing "CREATE TABLE t59489 (d12_3 DECIMAL(12,3), d4_2 DECIMAL(4,2))"
    comparator.go:98: Executing "INSERT INTO t59489(d12_3) VALUES (6000) RETURNING d12_3"
    comparator.go:98: Executing "INSERT INTO t59489(d12_3) VALUES (6e3) RETURNING d12_3"
    comparator.go:98: Executing "SELECT d12_3 FROM t59489"
    comparator.go:98: Executing "INSERT INTO t59489(d4_2) VALUES (600)"
    comparator.go:98: Executing "INSERT INTO t59489(d4_2) VALUES (6e2)"
    comparator.go:98: Executing "SELECT CASE WHEN false THEN 1::REGCLASS ELSE 2::REGNAMESPACE END; SELECT CASE WHEN false THEN 1::REGCLASS ELSE 2::REGPROC END; SELECT CASE WHEN false THEN 1::REGCLASS ELSE 2::REGPROCEDURE END; SELECT CASE WHEN false THEN 1::REGCLASS ELSE 2::REGROLE END; SELECT CASE WHEN false THEN 1::REGCLASS ELSE 2::REGTYPE END; SELECT CASE WHEN false THEN 1::REGNAMESPACE ELSE 2::REGCLASS END; SELECT CASE WHEN false THEN 1::REGNAMESPACE ELSE 2::REGPROC END; SELECT CASE WHEN false THEN 1::REGNAMESPACE ELSE 2::REGPROCEDURE END; SELECT CASE WHEN false THEN 1::REGNAMESPACE ELSE 2::REGROLE END; SELECT CASE WHEN false THEN 1::REGNAMESPACE ELSE 2::REGTYPE END; SELECT CASE WHEN false THEN 1::REGPROC ELSE 2::REGCLASS END; SELECT CASE WHEN false THEN 1::REGPROC ELSE 2::REGNAMESPACE END; SELECT CASE WHEN false THEN 1::REGPROC ELSE 2::REGPROCEDURE END; SELECT CASE WHEN false THEN 1::REGPROC ELSE 2::REGROLE END; SELECT CASE WHEN false THEN 1::REGPROC ELSE 2::REGTYPE END; SELECT CASE WHEN false THEN 1::REGPROCEDURE ELSE 2::REGCLASS END; SELECT CASE WHEN false THEN 1::REGPROCEDURE ELSE 2::REGNAMESPACE END; SELECT CASE WHEN false THEN 1::REGPROCEDURE ELSE 2::REGPROC END; SELECT CASE WHEN false THEN 1::REGPROCEDURE ELSE 2::REGROLE END; SELECT CASE WHEN false THEN 1::REGPROCEDURE ELSE 2::REGTYPE END; SELECT CASE WHEN false THEN 1::REGROLE ELSE 2::REGCLASS END; SELECT CASE WHEN false THEN 1::REGROLE ELSE 2::REGNAMESPACE END; SELECT CASE WHEN false THEN 1::REGROLE ELSE 2::REGPROC END; SELECT CASE WHEN false THEN 1::REGROLE ELSE 2::REGPROCEDURE END; SELECT CASE WHEN false THEN 1::REGROLE ELSE 2::REGTYPE END; SELECT CASE WHEN false THEN 1::REGTYPE ELSE 2::REGCLASS END; SELECT CASE WHEN false THEN 1::REGTYPE ELSE 2::REGNAMESPACE END; SELECT CASE WHEN false THEN 1::REGTYPE ELSE 2::REGPROC END; SELECT CASE WHEN false THEN 1::REGTYPE ELSE 2::REGPROCEDURE END; SELECT CASE WHEN false THEN 1::REGTYPE ELSE 2::REGROLE END"
    comparator.go:98: Executing "SELECT (1 / 3.0)::FLOAT4::STRING, (-1 / 3.0)::FLOAT8::STRING"
    comparator.go:98: Executing "SET extra_float_digits = -15"
    comparator.go:98: Executing "SELECT (-1 / 3.0)::FLOAT4::STRING, (1 / 3.0)::FLOAT8::STRING"
    comparator.go:98: Executing "CREATE TABLE def_assn_cast (id INT4, a INT4 DEFAULT 1.0::FLOAT4, b VARCHAR DEFAULT 'true'::BOOL, c NAME DEFAULT 'foo'::CHAR)"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32261

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12148802?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12148802?buildTab=artifacts#/) on release-23.2 @ [fe7abe63ff6ac18b59971c5c6e21f43f4e8b5f5a](https://github.com/cockroachdb/cockroach/commits/fe7abe63ff6ac18b59971c5c6e21f43f4e8b5f5a):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/gen_test_objects (20m48s)

```
Stack: 

```
goroutine 4961013 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true}'::JSONB), ARRAY['seed'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123,\"name_gen\":{\"noise\":false}}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123,\"name_gen\":{\"number\":false}}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123,\"name_gen\":{\"number\":false,\"noise\":false}}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123,\"name_gen\":{\"number\":false,\"noise\":false,\"quote\":1}}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123,\"name_gen\":{\"noise\":false,\"zalgo\":true}}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT json_remove_path(crdb_internal.generate_test_objects('{\"dry_run\":true,\"seed\":123,\"name_gen\":{\"noise\":true,\"zalgo\":true}}'::JSONB), ARRAY['generated_counts'])"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('foo.bar.baz', ARRAY[0, 10, 20])->'generated_counts'"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"names\":\"dba.bar.baz\", \"counts\":[2,0,10], \"name_gen\":{\"noise\":false}}'::JSONB)->'generated_counts'"
    comparator.go:98: Executing "SELECT count(*) FROM [SHOW TABLES FROM dba1]"
    comparator.go:98: Executing "SELECT count(*) FROM [SHOW TABLES FROM dba2]"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"names\":\"dbb.bar.baz\", \"counts\":[1,1,0], \"name_gen\":{\"noise\":false}}'::JSONB)->'generated_counts'"
    comparator.go:98: Executing "SELECT schema_name FROM [SHOW SCHEMAS FROM dbb1]"
    comparator.go:98: Executing "SELECT count(*) FROM [SHOW TABLES FROM dbb1.bar1]"
    comparator.go:98: Executing "CREATE SCHEMA myschema"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"names\":\"myschema.foo\", \"counts\":[2], \"name_gen\":{\"noise\":false}}'::JSONB)->'generated_counts'"
    comparator.go:98: Executing "SELECT schema_name, table_name FROM [SHOW TABLES FROM test.myschema]"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('myschema.foo', ARRAY[0, 2])->'generated_counts'"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('myschema.foo', ARRAY[0])->'generated_counts'"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"names\":\"scgen.foo\", \"counts\":[2,0], \"name_gen\":{\"noise\":false}}'::JSONB)->'generated_counts'"
    comparator.go:98: Executing "SELECT schema_name FROM [SHOW SCHEMAS] WHERE schema_name LIKE 'scgen%'"
    comparator.go:98: Executing "SELECT count(*) FROM [SHOW TABLES FROM scgen1]"
    comparator.go:98: Executing "SELECT count(*) FROM [SHOW TABLES FROM scgen2]"
    comparator.go:98: Executing "CREATE SCHEMA otherschema; SET search_path = invalidschema, otherschema, public"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"names\":\"foo\", \"counts\":[2], \"name_gen\":{\"noise\":false}}'::JSONB)->'generated_counts'"
    comparator.go:98: Executing "SELECT schema_name, table_name FROM [SHOW TABLES FROM test.otherschema]"
    comparator.go:98: Executing "RESET search_path"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"names\":\"dbfoo.baz\", \"counts\":[1,0,2], \"name_gen\":{\"noise\":false}}'::JSONB)->'generated_counts'"
    comparator.go:98: Executing "SELECT schema_name, table_name FROM [SHOW TABLES FROM dbfoo1]"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('foo', ARRAY[10, 10, 10])"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('foo', ARRAY[10, 10])"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('nonexistent.foo.bar', 0)"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('nonexistent.bar', 0)"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('test.nonexistent.bar', 0)"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('pg_foo.bar', ARRAY[10, 10])"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('test.pg_foo.bar', ARRAY[10, 10])"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('system.public.foo', 10)"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('pg_catalog.foo', 10)"
    comparator.go:98: Executing "SET search_path = pg_catalog, public"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('foo', 10)"
    comparator.go:98: Executing "RESET search_path"
    comparator.go:98: Executing "CREATE DATABASE rootonly; CREATE TABLE rootonly.foo (x INT8); CREATE TABLE rootonly.bar (x INT8)"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('foo._._', ARRAY[1, 0, 0])"
    comparator.go:98: Executing "SET CLUSTER SETTING \"sql.schema.test_object_generator.non_admin.enabled\" = true"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('foo._._', ARRAY[1, 0, 0])"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"table_templates\":[\"rootonly.foo\"]}'::JSONB)"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('{\"table_templates\":[\"rootonly.*\"]}'::JSONB)"
    comparator.go:98: Executing "ALTER USER testuser WITH CREATEDB"
    comparator.go:98: Executing "SELECT crdb_internal.generate_test_objects('custom._._', ARRAY[1, 0, 0])"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12176652?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12176652?buildTab=artifacts#/) on release-23.2 @ [abd56c6d6e9dbf131fc1fa0d3f8d1fbcfb7e4901](https://github.com/cockroachdb/cockroach/commits/abd56c6d6e9dbf131fc1fa0d3f8d1fbcfb7e4901):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/new_schema_changer (7s)

```
Stack: 

```
goroutine 7216668 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:96: Executing "SHOW GRANTS ON TABLE v"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE v FOR readwrite, \"test-user\""
    comparator.go:96: Executing "SHOW GRANTS FOR readwrite, \"test-user\""
    comparator.go:96: Executing "REVOKE ALL ON TABLE v FROM readwrite, \"test-user\""
    comparator.go:96: Executing "SHOW GRANTS ON TABLE v"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE v FOR readwrite, \"test-user\""
    comparator.go:96: Executing "SHOW GRANTS FOR readwrite, \"test-user\""
    comparator.go:96: Executing "SHOW GRANTS ON DATABASE a"
    comparator.go:96: Executing "SET database = \"\""
    comparator.go:96: Executing "GRANT ALL ON a.t@xyz TO readwrite\n"
    comparator.go:96: Executing "GRANT ALL ON TABLE * TO readwrite"
    comparator.go:96: Executing "GRANT ALL ON TABLE a.t, a.tt TO readwrite"
    comparator.go:96: Executing "GRANT ALL ON DATABASE * TO readwrite\n"
    comparator.go:96: Executing "CREATE DATABASE b"
    comparator.go:96: Executing "CREATE TABLE b.t (id INT8 PRIMARY KEY)"
    comparator.go:96: Executing "CREATE TABLE b.t2 (id INT8 PRIMARY KEY)"
    comparator.go:96: Executing "CREATE DATABASE c"
    comparator.go:96: Executing "CREATE TABLE c.t (id INT8 PRIMARY KEY)"
    comparator.go:96: Executing "SET database = b"
    comparator.go:96: Executing "GRANT ALL ON TABLE * TO vanilli"
    comparator.go:96: Executing "CREATE USER vanilli"
    comparator.go:96: Executing "GRANT ALL ON TABLE * TO vanilli"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE *"
    comparator.go:96: Executing "CREATE USER millie"
    comparator.go:96: Executing "GRANT ALL ON TABLE c.*, b.t TO millie"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE b.*"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE a.*, b.*"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE c.t"
    comparator.go:96: Executing "REVOKE ALL ON TABLE *, c.* FROM vanilli"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE b.*"
    comparator.go:96: Executing "CREATE DATABASE empty"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE empty.*"
    comparator.go:96: Executing "SHOW GRANTS ON TABLE empty.*, b.*"
    comparator.go:96: Executing "GRANT USAGE ON TABLE t TO testuser"
    comparator.go:96: Executing "GRANT SELECT ON TABLE system.lease TO testuser"
    comparator.go:96: Executing "REVOKE SELECT ON TABLE system.lease FROM testuser"
    comparator.go:96: Executing "GRANT RULE ON TABLE t TO testuser"
    comparator.go:96: Executing "GRANT SELECT ON TABLE empty.* TO testuser"
    comparator.go:96: Executing "CREATE SCHEMA empty.sc"
    comparator.go:96: Executing "GRANT SELECT ON TABLE empty.sc.* TO testuser"
    comparator.go:96: Executing "CREATE DATABASE db; SET database = db; CREATE TABLE t (i INT8)"
    schemachanger_test.go:935: schema changer comparator testing succeeded on logic test "grant_table"
=== RUN   TestComparatorFromLogicTests/new_schema_changer
    schemachanger_test.go:930: running schema changer comparator testing on statements collected from logic test "new_schema_changer"
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    comparator.go:96: Executing "CREATE DATABASE IF NOT EXISTS test"
    comparator.go:96: Executing "CREATE USER testuser"
    comparator.go:96: Executing ""
    comparator.go:96: Executing "CREATE TABLE foo (i INT8 PRIMARY KEY)"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12229640?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12229640?buildTab=artifacts#/) on release-23.2 @ [c5dc7554124b2de67f9db3e5f37b82037859f5b7](https://github.com/cockroachdb/cockroach/commits/c5dc7554124b2de67f9db3e5f37b82037859f5b7):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/trigram_builtins (6s)

```
Stack: 

```
goroutine 6724061 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:96: Executing "SELECT (x).* FROM (SELECT unnest(ARRAY[t58438.*]) FROM t58438) AS v (x)"
    comparator.go:96: Executing "SET testing_optimizer_disable_rule_probability = 1.0"
    comparator.go:96: Executing "CREATE TABLE t95615 (c1 INET PRIMARY KEY)"
    comparator.go:96: Executing "INSERT INTO t95615 VALUES ('192.168.0.1'::INET)"
    comparator.go:96: Executing "SELECT * FROM t95615, LATERAL ROWS FROM (hostmask(c1), hostmask(c1)) WITH ORDINALITY"
    comparator.go:96: Executing "CREATE FUNCTION f95615()\n\tRETURNS FLOAT8\n\tIMMUTABLE\n\tLEAKPROOF\n\tLANGUAGE SQL\n\tAS $funcbody$SELECT 1.1$funcbody$"
    comparator.go:96: Executing "SELECT * FROM t95615, LATERAL ROWS FROM (f95615(), f95615()) WITH ORDINALITY"
    comparator.go:96: Executing "RESET testing_optimizer_disable_rule_probability"
    comparator.go:96: Executing "CREATE TABLE t95315 (c1 INT8, c2 FLOAT8)"
    comparator.go:96: Executing "INSERT INTO t95315 VALUES (1, 4.3), (2, 5.4), (3, 6), (4, 7)"
    comparator.go:96: Executing "SELECT 1, c1, c2 FROM t95315 JOIN ROWS FROM (CAST(c1 AS INT8), CAST(c2 AS INT8)) USING (c1, c2) ORDER BY 1, 2, 3"
    comparator.go:96: Executing "SELECT CASE WHEN true THEN (SELECT * FROM ROWS FROM (generate_series(1, 3)) LIMIT 1) ELSE 1 END"
    comparator.go:96: Executing "SELECT COALESCE((SELECT * FROM ROWS FROM (generate_series(1, 3)) LIMIT 1))"
    comparator.go:96: Executing "SELECT CASE WHEN true THEN sum(x) ELSE 1 END FROM xy"
    comparator.go:96: Executing "SELECT COALESCE(sum(x)) FROM xy"
    comparator.go:96: Executing "SELECT CASE WHEN true THEN sum(x) OVER () ELSE 1 END FROM xy"
    comparator.go:96: Executing "SELECT COALESCE(sum(x) OVER ()) FROM xy"
    comparator.go:96: Executing "SELECT NULLIF(generate_series(1, x), generate_series(1, 3)) FROM xy"
    comparator.go:96: Executing "SELECT unnest('{}')"
    comparator.go:96: Executing "SELECT unnest('{}', '{}', '{}')"
    comparator.go:96: Executing "SELECT unnest(1)"
    comparator.go:96: Executing "SELECT unnest(1, 2, 3)"
    comparator.go:96: Executing "SELECT unnest(NULL)"
    comparator.go:96: Executing "SELECT unnest(NULL, NULL, NULL)"
    comparator.go:96: Executing "SELECT information_schema._pg_expandarray('{}')"
    comparator.go:96: Executing "SELECT unnest('{1}'::INT8[], '{2}', '{3}')"
    comparator.go:96: Executing "SELECT unnest('{1}'::INT8[], '{}', '{}')"
    comparator.go:96: Executing "SELECT unnest('{1}', '{2}', '{3}'::INT8[])"
    comparator.go:96: Executing "SELECT unnest('{}', '{}', '{3}'::INT8[])"
    schemachanger_test.go:935: schema changer comparator testing succeeded on logic test "srfs"
=== RUN   TestComparatorFromLogicTests/trigram_builtins
    schemachanger_test.go:930: running schema changer comparator testing on statements collected from logic test "trigram_builtins"
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    comparator.go:96: Executing "CREATE DATABASE IF NOT EXISTS test"
    comparator.go:96: Executing "CREATE USER testuser"
    comparator.go:96: Executing "SELECT show_trgm(str) FROM (VALUES (''), ('a'), ('ab'), ('abc'), ('abcd'), (e'\\u041F\\u0440\\u0438\\u0432\\u0435\\u0442hi'), (NULL)) AS tbl (str)"
    comparator.go:96: Executing "SELECT show_trgm('dcba')"
    comparator.go:96: Executing "SELECT show_trgm('Foo')"
    comparator.go:96: Executing "SELECT show_trgm('aaaa')"
    comparator.go:96: Executing "SELECT show_trgm('a b,c|d_e3 bl  ')"
    comparator.go:96: Executing "SELECT similarity(a, b) FROM (VALUES ('', ''), ('foo', ''), ('', 'foo'), ('foo', NULL), (NULL, 'foo'), (NULL, NULL), ('foo', 'bar'), ('foo', 'far'), ('foo', 'for'), ('foo', 'foo')) AS tbl (a, b)"
    comparator.go:96: Executing "SELECT similarity(a, b) FROM (VALUES ('f', 'bfr'), ('foo', 'foobar'), ('', 'blah'), ('blah', '')) AS tbl (a, b)"
    comparator.go:96: Executing "SELECT similarity(a, b) FROM (VALUES ('FOO', 'foo'), ('foobar', 'foo'), ('foobar', 'barfoo'), ('blorp', 'z')) AS tbl (a, b)"
    comparator.go:96: Executing "SELECT similarity(a, b) FROM (VALUES ('foo', 'foobar'), ('foobar', 'foo'), ('FOO', 'foo'), ('foo', 'FOO'), ('blorp', 'z'), ('z', 'blorp')) AS tbl (a, b)"
    comparator.go:96: Executing "SHOW \"pg_trgm.similarity_threshold\""
    comparator.go:96: Executing "SELECT 'FOO' % 'foo', 'foobar' % 'foo', 'foobar' % 'barfoo', 'blorp' % 'z'"
    comparator.go:96: Executing "SET \"pg_trgm.similarity_threshold\" = .1"
    comparator.go:96: Executing "SELECT 'FOO' % 'foo', 'foobar' % 'foo', 'foobar' % 'barfoo', 'blorp' % 'z'"
    comparator.go:96: Executing "SET \"pg_trgm.similarity_threshold\" = 1"
    comparator.go:96: Executing "SELECT 'FOO' % 'foo', 'foobar' % 'foo', 'foobar' % 'barfoo', 'blorp' % 'z'"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12247730?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12247730?buildTab=artifacts#/) on release-23.2 @ [3744e5655559d3350aedaf1fbbaecbf1e69c98d6](https://github.com/cockroachdb/cockroach/commits/3744e5655559d3350aedaf1fbbaecbf1e69c98d6):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/plpgsql_cursor (50s)

```
Stack: 

```
goroutine 6846265 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f(IN n INT8)\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx INT;\ny INT;\ni INT := 0;\nBEGIN\nOPEN curs FOR SELECT * FROM xy ORDER BY x;\nWHILE i < n LOOP\nFETCH curs INTO x, y;\nRAISE NOTICE 'i: %, x: %, y: %', i, x, y;\ni := i + 1;\nEND LOOP;\nRETURN 0;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f(0)"
    comparator.go:94: Executing "SELECT f(1)"
    comparator.go:94: Executing "SELECT f(2)"
    comparator.go:94: Executing "SELECT f(3)"
    comparator.go:94: Executing "SELECT f(4)"
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f()\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx INT;\ni INT := 0;\nBEGIN\nOPEN curs FOR SELECT * FROM xy ORDER BY x;\nWHILE i < 5 LOOP\nFETCH curs INTO x;\nRAISE NOTICE 'i: %, x: %', i, x;\ni := i + 1;\nEND LOOP;\nRETURN 0;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f()"
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f()\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx INT;\ny INT := 1000;\ni INT := 0;\nBEGIN\nOPEN curs FOR SELECT x FROM xy ORDER BY x;\nWHILE i < 5 LOOP\nFETCH curs INTO x, y;\nRAISE NOTICE 'i: %, x: %, y: %', i, x, y;\ni := i + 1;\nEND LOOP;\nRETURN 0;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f()"
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f_fetch(IN curs REFCURSOR)\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\nx INT;\ny INT;\nBEGIN\nFETCH curs INTO x, y;\nRAISE NOTICE 'x: %, y: %', x, y;\nRETURN 0;\nEND\n$funcbody$; CREATE OR REPLACE FUNCTION f_move(IN curs REFCURSOR)\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nBEGIN\nMOVE curs;\nRETURN 0;\nEND\n$funcbody$; BEGIN TRANSACTION"
    comparator.go:94: Executing "INSERT INTO xy VALUES (7, 8), (9, 10);\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT f_move('foo');\n"
    comparator.go:94: Executing "SELECT f_fetch('foo');\n"
    comparator.go:94: Executing "SELECT f_move('foo');\nSELECT f_move('foo');\n"
    comparator.go:94: Executing "SELECT f_fetch('foo');\n"
    comparator.go:94: Executing "SELECT f_move('foo');\nSELECT f_move('foo');\n"
    comparator.go:94: Executing "SELECT f_fetch('foo');\n"
    comparator.go:94: Executing "ABORT;\nCREATE OR REPLACE FUNCTION f(n INT) RETURNS INT AS $$\nDECLARE\ncurs REFCURSOR := format('foo%s', n)::REFCURSOR;\nx INT;\nBEGIN\nOPEN curs FOR SELECT * FROM xy ORDER BY x;\nIF n = 0 THEN\nFETCH curs INTO x;\nELSIF n = 1 THEN\nFETCH NEXT curs INTO x;\nELSIF n = 2 THEN\nFETCH PRIOR curs INTO x;\nELSIF n = 3 THEN\nFETCH FIRST curs INTO x;\nELSIF n = 4 THEN\nFETCH LAST curs INTO x;\nELSIF n = 5 THEN\nFETCH ABSOLUTE 2 curs INTO x;\nELSIF n = 6 THEN\nFETCH ABSOLUTE -2 curs INTO x;\nELSIF n = 7 THEN\nFETCH RELATIVE 2 curs INTO x;\nELSIF n = 8 THEN\nFETCH RELATIVE -2 curs INTO x;\nELSIF n = 9 THEN\nFETCH FORWARD 2 curs INTO x;\nELSIF n = 10 THEN\nFETCH BACKWARD 2 curs INTO x;\nEND IF;\nRETURN x;\nEND\n$$ LANGUAGE PLpgSQL;\n"
    comparator.go:94: Executing "SELECT f(0), f(1), f(3), f(5), f(7), f(9)"
    comparator.go:94: Executing "SELECT f(2)"
    comparator.go:94: Executing "SELECT f(4)"
    comparator.go:94: Executing "SELECT f(6)"
    comparator.go:94: Executing "SELECT f(8)"
    comparator.go:94: Executing "SELECT f(10)"
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f(IN n INT8)\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := format('foo%s', n)::REFCURSOR;\nx INT;\nBEGIN\nOPEN curs FOR SELECT * FROM xy ORDER BY x;\nIF n = 0 THEN\nMOVE curs;\nELSIF n = 1 THEN\nMOVE NEXT curs;\nELSIF n = 2 THEN\nMOVE PRIOR curs;\nELSIF n = 3 THEN\nMOVE FIRST curs;\nELSIF n = 4 THEN\nMOVE LAST curs;\nELSIF n = 5 THEN\nMOVE ABSOLUTE 2 curs;\nELSIF n = 6 THEN\nMOVE ABSOLUTE -2 curs;\nELSIF n = 7 THEN\nMOVE RELATIVE 2 curs;\nELSIF n = 8 THEN\nMOVE RELATIVE -2 curs;\nELSIF n = 9 THEN\nMOVE FORWARD 2 curs;\nELSIF n = 10 THEN\nMOVE BACKWARD 2 curs;\nELSIF n = 11 THEN\nMOVE FORWARD ALL curs;\nELSIF n = 12 THEN\nMOVE BACKWARD ALL curs;\nELSIF n = 13 THEN\nMOVE ALL curs;\nEND IF;\nFETCH curs INTO x;\nRETURN x;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f(0), f(1), f(3), f(5), f(7), f(9), f(11), f(13)"
    comparator.go:94: Executing "SELECT f(2)"
    comparator.go:94: Executing "SELECT f(4)"
    comparator.go:94: Executing "SELECT f(6)"
    comparator.go:94: Executing "SELECT f(8)"
    comparator.go:94: Executing "SELECT f(10)"
    comparator.go:94: Executing "SELECT f(12)"
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f(IN n INT8)\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx INT;\nBEGIN\nIF n = 0 THEN\nFETCH FIRST curs INTO x;\nELSE\nMOVE FIRST curs;\nEND IF;\nRETURN x;\nEND\n$funcbody$"
    comparator.go:94: Executing "BEGIN TRANSACTION; DECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x"
    comparator.go:94: Executing "SELECT f(1);\nSELECT f(1);\nSELECT f(1);\n"
    comparator.go:94: Executing "SELECT f(0);\n"
    comparator.go:94: Executing "SELECT f(1);\nSELECT f(1);\n"
    comparator.go:94: Executing "SELECT f(0);\n"
    comparator.go:94: Executing "FETCH FORWARD 2 FROM foo;\n"
    comparator.go:94: Executing "SELECT f(1);\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x;\nMOVE FORWARD 2 IN foo;\n"
    comparator.go:94: Executing "SELECT f(0);\n"
    comparator.go:94: Executing "ABORT;\n"
    comparator.go:94: Executing "CREATE OR REPLACE FUNCTION f()\n\tRETURNS INT8\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx INT;\nBEGIN\nOPEN curs FOR SELECT '1';\nFETCH curs INTO x;\nRETURN x;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f()"
    comparator.go:94: Executing "DROP FUNCTION f(); CREATE OR REPLACE FUNCTION f()\n\tRETURNS BOOL\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx BOOL;\nBEGIN\nOPEN curs FOR SELECT 'f';\nFETCH curs INTO x;\nRETURN x;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f()"
    comparator.go:94: Executing "DROP FUNCTION f(); CREATE OR REPLACE FUNCTION f()\n\tRETURNS CHAR\n\tLANGUAGE plpgsql\n\tAS $funcbody$\nDECLARE\ncurs REFCURSOR := 'foo';\nx CHAR;\nBEGIN\nOPEN curs FOR SELECT 'a';\nFETCH curs INTO x;\nRETURN x;\nEND\n$funcbody$"
    comparator.go:94: Executing "SELECT f()"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12265160?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12265160?buildTab=artifacts#/) on release-23.2 @ [935399600693f84fb746715f430ecc245934f3cd](https://github.com/cockroachdb/cockroach/commits/935399600693f84fb746715f430ecc245934f3cd):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/aggregate (3m1s)

```
Stack: 

```
goroutine 6314994 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:94: Executing "DELETE FROM vals"
    comparator.go:94: Executing "INSERT INTO vals VALUES (NULL::VARBIT), (NULL::VARBIT), (NULL::VARBIT), (NULL::VARBIT)"
    comparator.go:94: Executing "SELECT bit_and(v) FROM vals"
    comparator.go:94: Executing "SELECT bit_or(v) FROM vals"
    comparator.go:94: Executing "SELECT bit_and(NULL)"
    comparator.go:94: Executing "SELECT bit_or(NULL)"
    comparator.go:94: Executing "SELECT bit_and(x::VARBIT) FROM (VALUES ('1'), ('11')) AS t (x)"
    comparator.go:94: Executing "SELECT bit_and(x) FROM (VALUES ('100'::BIT(3)), ('101010111'::VARBIT)) AS t (x)"
    comparator.go:94: Executing "SELECT bit_and(x) FROM (VALUES (''::VARBIT), ('1'::VARBIT)) AS t (x)"
    comparator.go:94: Executing "SELECT bit_or(x::VARBIT) FROM (VALUES ('1'), ('11')) AS t (x)"
    comparator.go:94: Executing "SELECT bit_or(x) FROM (VALUES ('100'::BIT(3)), ('101010111'::VARBIT)) AS t (x)"
    comparator.go:94: Executing "SELECT bit_or(x) FROM (VALUES (''::VARBIT), ('1'::VARBIT)) AS t (x)"
    comparator.go:94: Executing "CREATE TABLE t46981_0 (c0 INT8); CREATE VIEW v46981_0 (c0) AS SELECT count_rows() FROM t46981_0"
    comparator.go:94: Executing "SELECT * FROM v46981_0 WHERE '' !~ '+'"
    comparator.go:94: Executing "DROP TABLE IF EXISTS osagg"
    comparator.go:94: Executing "CREATE TABLE osagg (f FLOAT8, s STRING, i INTERVAL)"
    comparator.go:94: Executing "INSERT INTO osagg VALUES (NULL, NULL, NULL), (0.00, NULL, '1 months'), (0.05, NULL, '1 months'), (1.0, 'v1', '1 year 1 months'), (3.0, 'v3', '1 year 3 months'), (5.0, 'v5', '1 year 5 months'), (2.0, 'v2', '1 year 2 months'), (4.0, 'v4', '1 year 4 months'), (6.0, 'v6', '1 year 6 months')"
    comparator.go:94: Executing "SELECT percentile_disc(0.95) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(0.95) WITHIN GROUP (ORDER BY f), percentile_disc(0.95) WITHIN GROUP (ORDER BY s) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_cont(0.95) WITHIN GROUP (ORDER BY f), percentile_cont(0.95) WITHIN GROUP (ORDER BY f DESC), percentile_cont(0.95) WITHIN GROUP (ORDER BY i) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(0.00) WITHIN GROUP (ORDER BY s), percentile_disc(0.1) WITHIN GROUP (ORDER BY f), percentile_disc(0.15) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_cont(0.05) WITHIN GROUP (ORDER BY f), percentile_cont(0.05) WITHIN GROUP (ORDER BY f DESC), percentile_cont(0.05) WITHIN GROUP (ORDER BY i), percentile_cont(0.05) WITHIN GROUP (ORDER BY i DESC) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(0.25) WITHIN GROUP (ORDER BY f), percentile_disc(0.5) WITHIN GROUP (ORDER BY f), percentile_disc(0.75) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_cont(0.25) WITHIN GROUP (ORDER BY f), percentile_cont(0.5) WITHIN GROUP (ORDER BY f), percentile_cont(0.75) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(ARRAY[0.25]::FLOAT8[]) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(ARRAY[0.25, 0.5, 0.75]::FLOAT8[]) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_cont(ARRAY[0.25, 0.5, 0.75]::FLOAT8[]) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(ARRAY[0.25, 0.5, 0.75]::FLOAT8[]) WITHIN GROUP (ORDER BY i) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(ARRAY[1.25]::FLOAT8[]) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_disc(ARRAY[0.25, 0.50, 1.25]::FLOAT8[]) WITHIN GROUP (ORDER BY f) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_cont(ARRAY[0.25, 0.5, 0.75]::FLOAT8[]) WITHIN GROUP (ORDER BY i) FROM osagg"
    comparator.go:94: Executing "CREATE VIEW osagg_view (disc, cont) AS SELECT percentile_disc(0.50) WITHIN GROUP (ORDER BY f), percentile_cont(0.50) WITHIN GROUP (ORDER BY f DESC) FROM osagg"
    comparator.go:94: Executing "SHOW CREATE osagg_view"
    comparator.go:94: Executing "SELECT percentile_disc(0.50) FROM osagg"
    comparator.go:94: Executing "SELECT percentile_cont(0.50) FROM osagg"
    comparator.go:94: Executing "CREATE TABLE t_collate (x STRING COLLATE en_US); INSERT INTO t_collate VALUES ('hi' COLLATE en_US), ('hello' COLLATE en_US), ('howdy' COLLATE en_US)"
    comparator.go:94: Executing "SELECT min(x), max(x) FROM t_collate"
    comparator.go:94: Executing "SELECT min(NULL::STRING COLLATE en_US), max(NULL::STRING COLLATE en_US)"
    comparator.go:94: Executing "CREATE TABLE profiles (userid INT8 NOT NULL, property STRING NOT NULL, value STRING NOT NULL)"
    comparator.go:94: Executing "INSERT INTO profiles VALUES (1, 'email', 'user1@gmail.com'), (1, 'phone', '111111111111111'), (1, 'home_page', 'user1.org1.com'), (2, 'email', 'user2@gmail.com'), (2, 'phone', '222222222222222'), (2, 'home_page', 'user2.org1.com')"
    comparator.go:94: Executing "SELECT json_object_agg(property, value) FROM profiles GROUP BY userid ORDER BY userid"
    comparator.go:94: Executing "CREATE TABLE users (userid INT8 PRIMARY KEY, user_name STRING NOT NULL)"
    comparator.go:94: Executing "CREATE TABLE user_networks (userid INT8 NOT NULL REFERENCES users (userid), network_name STRING NOT NULL, user_account_id STRING NOT NULL)"
    comparator.go:94: Executing "INSERT INTO users VALUES (1, 'Alice'), (2, 'Bob')"
    comparator.go:94: Executing "INSERT INTO user_networks VALUES ('1', 'Facebook', 'Alice_fb'), ('1', 'Twitter', '@Alice'), ('1', 'Instagram', 'AliceInst'), ('2', 'Facebook', 'Bob_fb'), ('2', 'LinkedIn', 'Bob The Builder')"
    comparator.go:94: Executing "SELECT json_object_agg(user_name, networks) FROM (SELECT u.userid AS userid, u.user_name, json_object_agg(un.network_name, un.user_account_id) AS networks FROM users AS u, user_networks AS un WHERE u.userid = un.userid GROUP BY u.userid ORDER BY u.userid) GROUP BY userid ORDER BY userid"
    comparator.go:94: Executing "SELECT json_object_agg(NULL, NULL)"
    comparator.go:94: Executing "SELECT json_object_agg(NULL, 1)"
    comparator.go:94: Executing "CREATE TABLE persons (id STRING PRIMARY KEY, name STRING NOT NULL)"
    comparator.go:94: Executing "CREATE TABLE companies (id STRING PRIMARY KEY, company_name STRING NOT NULL)"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12281336?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12281336?buildTab=artifacts#/) on release-23.2 @ [dc24db6b7ed2f53d034ae9b8038ce08fc7d57542](https://github.com/cockroachdb/cockroach/commits/dc24db6b7ed2f53d034ae9b8038ce08fc7d57542):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/procedure_schema_change (15s)

```
Stack: 

```
goroutine 4963205 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::DECIMAL,))).*;\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT 5.0101::FLOAT FROM generate_series(1, 5);\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::DECIMAL,))).*;\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::INT,))).*;\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::STRING,))).*;\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT 5.0101::DECIMAL FROM generate_series(1, 5);\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::FLOAT,))).*;\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::INT,))).*;\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::STRING,))).*;\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT 'abc';\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::CHAR,))).*;\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT '101';\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::BIT,))).*;\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT 10000000;\n"
    comparator.go:94: Executing "SELECT (crdb_internal.plpgsql_fetch('foo', 0, 1, (NULL::INT2,))).*;\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nINSERT INTO xy VALUES (7, 8), (9, 10), (11, 12), (13, 14), (15, 16);\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 3, 0, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 0, 2, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 1, 3, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 2, 7, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 5, 0, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 0, -3, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 1, -3, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\nMOVE FORWARD 3 foo;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 2, 2, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\nMOVE FORWARD 3 foo;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 3, 0, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 4, 0, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 6, 0, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', -1, 0, (NULL::INT, NULL::INT));\n"
    comparator.go:94: Executing "ABORT;\nBEGIN;\nDECLARE foo CURSOR FOR SELECT * FROM xy ORDER BY x DESC;\n"
    comparator.go:94: Executing "SELECT crdb_internal.plpgsql_fetch('foo', 10, 0, (NULL::INT, NULL::INT));\n"
    schemachanger_test.go:941: schema changer comparator testing succeeded on logic test "plpgsql_builtins"
=== RUN   TestComparatorFromLogicTests/procedure_schema_change
    schemachanger_test.go:936: running schema changer comparator testing on statements collected from logic test "procedure_schema_change"
    comparator.go:94: Executing "CREATE DATABASE IF NOT EXISTS test"
    comparator.go:94: Executing "CREATE USER testuser"
    comparator.go:94: Executing "CREATE PROCEDURE p()\n\tLANGUAGE SQL\n\tAS $funcbody$SELECT 1$funcbody$"
    comparator.go:94: Executing "CREATE PROCEDURE p2()\n\tLANGUAGE SQL\n\tAS $funcbody$SELECT 1$funcbody$"
    comparator.go:94: Executing "CREATE PROCEDURE p_int(IN INT8)\n\tLANGUAGE SQL\n\tAS $funcbody$SELECT 1$funcbody$"
    comparator.go:94: Executing "CREATE FUNCTION p_func()\n\tRETURNS INT8\n\tLANGUAGE SQL\n\tAS $funcbody$SELECT 1$funcbody$"
    comparator.go:94: Executing "ALTER PROCEDURE p RENAME TO p"
    comparator.go:94: Executing "ALTER PROCEDURE p RENAME TO p_func"
    comparator.go:94: Executing "ALTER FUNCTION p RENAME TO p_new"
    comparator.go:94: Executing "ALTER PROCEDURE p RENAME TO p2"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestComparatorFromLogicTests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12302452?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SchemaChangerComparatorFromSqlLogicStatementsBazel/12302452?buildTab=artifacts#/) on release-23.2 @ [dc24db6b7ed2f53d034ae9b8038ce08fc7d57542](https://github.com/cockroachdb/cockroach/commits/dc24db6b7ed2f53d034ae9b8038ce08fc7d57542):

Fatal error:

```
panic: test timed out after 59m55s
running tests:
	TestComparatorFromLogicTests (59m55s)
	TestComparatorFromLogicTests/new_schema_changer (5m34s)

```
Stack: 

```
goroutine 6281322 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x3c5
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    comparator.go:94: Executing "CREATE TABLE p2 (i INT8 PRIMARY KEY, s STRING)"
    comparator.go:94: Executing ""
    comparator.go:94: Executing ""
    comparator.go:94: Executing "CREATE TYPE typ8 AS ENUM ('hello'); DROP TYPE typ8"
    comparator.go:94: Executing "CREATE TYPE typ8 AS ENUM ('hello')"
    comparator.go:94: Executing "CREATE TABLE t8 (x typ8)"
    comparator.go:94: Executing "DROP TYPE typ8"
    comparator.go:94: Executing "ALTER TABLE t8 ADD COLUMN y typ8"
    comparator.go:94: Executing "DROP TYPE typ8"
    comparator.go:94: Executing "ALTER TABLE t8 DROP COLUMN x"
    comparator.go:94: Executing "DROP TYPE typ8"
    comparator.go:94: Executing "ALTER TABLE t8 DROP COLUMN y"
    comparator.go:94: Executing "DROP TYPE typ8"
    comparator.go:94: Executing "CREATE TYPE typ8 AS ENUM ('hello'); ALTER TABLE t8 ADD COLUMN x typ8[]"
    comparator.go:94: Executing "DROP TYPE typ8"
    comparator.go:94: Executing "ALTER TABLE t8 DROP COLUMN x"
    comparator.go:94: Executing "DROP TYPE typ8"
    comparator.go:94: Executing "CREATE TYPE defaultdb.typ AS ENUM ('a')"
    comparator.go:94: Executing "CREATE TABLE defaultdb.ttyp (id INT8 PRIMARY KEY, name VARCHAR(256), x defaultdb.typ)"
    comparator.go:94: Executing "BEGIN TRANSACTION"
    comparator.go:94: Executing "DROP TABLE defaultdb.ttyp;\n"
    comparator.go:94: Executing "DROP TYPE defaultdb.typ;\n"
    comparator.go:94: Executing "SELECT count(*)\nFROM information_schema.tables\nWHERE table_schema = 'public'\nAND table_name LIKE '%typ%';\n"
    comparator.go:94: Executing "SELECT name, state\nFROM crdb_internal.tables\nWHERE schema_name = 'public'\nAND name LIKE '%typ%'\nORDER BY name;\n"
    comparator.go:94: Executing "COMMIT;\n"
    comparator.go:94: Executing "CREATE DATABASE db1"
    comparator.go:94: Executing "CREATE SCHEMA db1.sc1"
    comparator.go:94: Executing "CREATE SEQUENCE db1.sc1.sq1"
    comparator.go:94: Executing "CREATE TABLE db1.sc1.t1 (id INT8 PRIMARY KEY, name VARCHAR(256), val INT8 DEFAULT nextval('db1.sc1.sq1'))"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v1 AS (SELECT name FROM db1.sc1.t1)"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v2 AS (SELECT name AS n1, name AS n2 FROM db1.sc1.v1)"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v3 AS (SELECT name, n1 FROM db1.sc1.v1, db1.sc1.v2)"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v4 AS (SELECT n2, n1 FROM db1.sc1.v2)"
    comparator.go:94: Executing "CREATE TYPE db1.sc1.typ AS ENUM ('a')"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v5 AS (SELECT 'a'::db1.sc1.typ::STRING AS k, n2, n1 FROM db1.sc1.v4)"
    comparator.go:94: Executing "CREATE SCHEMA sc2"
    comparator.go:94: Executing "CREATE TYPE sc2.typ AS ENUM ('a')"
    comparator.go:94: Executing "DROP SCHEMA db1.sc1 CASCADE"
    comparator.go:94: Executing "DROP SCHEMA sc2 CASCADE"
    comparator.go:94: Executing "DROP DATABASE db1 CASCADE"
    comparator.go:94: Executing "CREATE ROLE test_set_role"
    comparator.go:94: Executing "CREATE DATABASE db1"
    comparator.go:94: Executing "ALTER ROLE test_set_role SET application_name = 'a'; ALTER ROLE test_set_role IN DATABASE db1 SET application_name = 'b'; ALTER ROLE ALL IN DATABASE db1 SET application_name = 'c'; ALTER ROLE ALL SET application_name = 'd'; ALTER ROLE test_set_role SET \"custom_option.setting\" = 'e'"
    comparator.go:94: Executing "CREATE SCHEMA db1.sc1"
    comparator.go:94: Executing "CREATE SEQUENCE db1.public.sq1"
    comparator.go:94: Executing "CREATE SEQUENCE db1.sc1.sq1"
    comparator.go:94: Executing "CREATE TABLE db1.sc1.t1 (id INT8 PRIMARY KEY, name VARCHAR(256), val INT8 DEFAULT nextval('db1.sc1.sq1'))"
    comparator.go:94: Executing "CREATE TABLE db1.public.t1 (id INT8 PRIMARY KEY, name VARCHAR(256), val INT8 DEFAULT nextval('db1.public.sq1'))"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v1 AS (SELECT name FROM db1.sc1.t1)"
    comparator.go:94: Executing "CREATE VIEW db1.sc1.v2 AS (SELECT name AS n1, name AS n2 FROM db1.sc1.v1)"
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111978 pkg/sql/schemachanger/schemachanger_test: TestComparatorFromLogicTests failed [C-test-failure O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComparatorFromLogicTests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/collatedstring
+++ b/pkg/sql/logictest/testdata/logic_test/collatedstring
@@ -363,8 +363,8 @@ SHOW CREATE TABLE quoted_coll
 quoted_coll  CREATE TABLE public.quoted_coll (
                a STRING COLLATE en NULL,
                b STRING COLLATE en_US NULL,
-               c STRING COLLATE en_US NULL DEFAULT 'c':::STRING COLLATE en_US,
-               d STRING COLLATE en_u_ks_level1 NULL DEFAULT 'd':::STRING COLLATE en_u_ks_level1,
+               c STRING COLLATE en_US NULL DEFAULT ('c':::STRING COLLATE en_US),
+               d STRING COLLATE en_u_ks_level1 NULL DEFAULT ('d':::STRING COLLATE en_u_ks_level1),
                e STRING COLLATE en_US NULL AS (a COLLATE en_US) STORED,
                rowid INT8 NOT VISIBLE NOT NULL DEFAULT unique_rowid(),
                CONSTRAINT quoted_coll_pkey PRIMARY KEY (rowid ASC)

--- a/pkg/sql/sem/tree/type_check.go
+++ b/pkg/sql/sem/tree/type_check.go
@@ -1580,8 +1580,15 @@ func (expr *ParenExpr) TypeCheck(
 	if err != nil {
 		return nil, err
 	}
-	// Parentheses are semantically unimportant and can be removed/replaced
-	// with its nested expression in our plan. This makes type checking cleaner.
+	// Parentheses are semantically unimportant and in most cases can be
+	// removed/replaced with its nested expression in our plan. This makes type
+	// checking cleaner. Collated string expressions need the parentheses to
+	// parse correctly in some cases.
+	if _, ok := exprTyped.(*CollateExpr); ok {
+		expr.Expr = exprTyped
+		expr.typ = exprTyped.ResolvedType()
+		return expr, nil
+	}
 	return exprTyped, nil
 }
 

--- a/pkg/sql/show_create_all_tables_builtin_test.go
+++ b/pkg/sql/show_create_all_tables_builtin_test.go
@@ -18,6 +18,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
 )
 
 // Use the output from crdb_internal.show_create_all_tables() to recreate the
@@ -36,22 +37,33 @@ func TestRecreateTables(t *testing.T) {
 	sqlRunner.Exec(t, `USE test;`)
 	sqlRunner.Exec(t, `CREATE TABLE foo(x INT primary key);`)
 	sqlRunner.Exec(t, `CREATE TABLE bar(x INT, y INT, z STRING, FAMILY f1(x, y, z))`)
+	sqlRunner.Exec(t, `
+		CREATE TABLE tab (
+			a STRING COLLATE en,
+			b STRING COLLATE en_US,
+			c STRING COLLATE en_US DEFAULT ('c' COLLATE en_US),
+			d STRING COLLATE en_u_ks_level1 DEFAULT ('d'::STRING COLLATE en_u_ks_level1),
+			e STRING COLLATE en_US AS (a COLLATE en_US) STORED,
+			f STRING COLLATE en_US ON UPDATE ('f' COLLATE en_US))`)
 
-	row := sqlRunner.QueryRow(t, "SELECT crdb_internal.show_create_all_tables('test')")
-	var recreateTablesStmt string
-	row.Scan(&recreateTablesStmt)
+	var recreateTablesStmts []string
+	for _, r := range sqlRunner.QueryStr(t, "SELECT crdb_internal.show_create_all_tables('test')") {
+		recreateTablesStmts = append(recreateTablesStmts, r[0])
+	}
 
 	// Use the recreateTablesStmt to recreate the tables, perform another
 	// show_create_all_tables and compare that the output is the same.
 	sqlRunner.Exec(t, `DROP DATABASE test;`)
 	sqlRunner.Exec(t, `CREATE DATABASE test;`)
-	sqlRunner.Exec(t, recreateTablesStmt)
 
-	row = sqlRunner.QueryRow(t, "SELECT crdb_internal.show_create_all_tables('test')")
-	var recreateTablesStmt2 string
-	row.Scan(&recreateTablesStmt2)
-
-	if recreateTablesStmt != recreateTablesStmt2 {
-		t.Fatalf("got: %s\nexpected: %s", recreateTablesStmt2, recreateTablesStmt)
+	for _, stmt := range recreateTablesStmts {
+		sqlRunner.Exec(t, stmt)
 	}
+
+	var recreateTablesStmts2 []string
+	for _, r := range sqlRunner.QueryStr(t, "SELECT crdb_internal.show_create_all_tables('test')") {
+		recreateTablesStmts2 = append(recreateTablesStmts2, r[0])
+	}
+
+	require.ElementsMatch(t, recreateTablesStmts, recreateTablesStmts2)
 }

